test/integration/command/source_test.py

```diff
@@ -142,6 +142,30 @@ def source(self):
         self.assertIn("conanfile.py: Running source!", client.out)
         self.assertEqual("Hello World", client.load("file1.txt"))
 
+    def test_local_source_layout_root(self):
```

Comment:
I think I'd change this test to implement the recipe that you provided, with the ``self.folders.root = ".."``. That helps to tag this as a ``Bugfix`` and be able to move it forward.

Then, I'd move new tests to cover the use case of a child "root" folder to a different PR, to be done later, once this is merged.